docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,169 @@
+
+### Alert Evaluation
+
+The Ruler itself has to communicate on 4 levels. All connections are presented in the diagram below.
+
+![ruler-connections-diagram](../assets/alerting-ruler.png)
+
+1. Alerts will be evaluated based on periodic requests to any replica of HTTP Query API, potentially through Query-frontend. 
+2. Ruler evaluates Alerts that can be triggered. Triggered alerts will be pushed to all replicas of the Alertmanager (discussed later) through AM HTTP API.
+3. It exposes gRPC Rules API which is consumed by Queriers. Querier than can show this API via it's HTTP Rule API.
+4. It saves samples to any replica of Receiver Router (or RouterIngestor). 
+
+### Rules API (non row one)
+
+We expect Rulers to be connected to Queriers through [gRPC Rules API](https://github.com/thanos-io/thanos/blob/main/pkg/rules/rulespb/rpc.proto#L29) (NOTE, not Store API as we have now). This allows queries to be able to present federated views of Recording and Alerting Rules loaded and active alerts through the [Promethus Rules HTTP API](https://prometheus.io/docs/prometheus/latest/querying/api/#rules). 
+
+This API is different but a bit overlapping with `/api/metrics/v1/{tenant}/api/v1/rules/raw` which represents what was configured (not loaded).
+
+In order to fulfil the full story we also need:
+
+* Implement matching / filtering by label/tenant on Rules HTTP API of Querier.
+
+This will allow users looking at the Thanos Rules API (or Query UI) to only see the alerts that matter to them.
+
+### Alert Evaluation Status
+
+On top of mentioned Rules API, Prometheus also exposes [HTTP Alert API](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts). We never expose it within Observatorium and Thanos and the truth is that Rules API gives exactly that information (on top of other data).
+
+Because of that we propose to not add this API, which will avoid a lot of work. In future iteration (e.g. if we would add UI), we could add Alert HTTP API in Observatorium which would extract data needed by this API from Rules API.
+
+### Alert Routing
+
+Triggered alerts are essential to be reliably forward to correct notifiers as soon as they are triggered. We propose deploying w-replica Alertmanager HA, through our configuration (without Prometheus Operator help which is usually used in our past deployments).
+
+We propose defining an additional HTTP API for routing on `/api/metrics/v1/{tenant}/api/v1/routing/raw` path in Observatorium API. Path that supports both GET, DELETE and PUT. 
+
+We propose to base the type on [Prometheus Operator AM Config](https://github.com/prometheus-operator/prometheus-operator/blob/9c0db5656f04e005de6a0413fd8eb8f11ec99757/pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go#L69). Proposed adding this API to upstream Alertmanager too (TODO).
```

Comment:
TODO: I will start a discussion on the Alertmanager repo.

TODO: Clarify what API should look like exactly, before merging

Reply:
I think we should clarify what exactly do we want here from the upstream Alertmanager. The fact is that AM already has an API  which would enable us to get currently loaded configuration, see the `/status` endpoint [in the spec](https://github.com/prometheus/alertmanager/blob/main/api/v2/openapi.yaml#L265).